### PR TITLE
8 adc bringup

### PR DESCRIPTION
**Pull request by jdo1156**
-Created ADC scripts
-Created MCP3008 Drivers
-Increased time between samples to 1 s

### CODE REVIEW

File: adc_manager/requirements/test
Comment (imcewan7529) [inline adc_manager/requirements/test:1]: Need to add a requirements.txt file here

File: adc_manager/src/adc.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    value = adc.read(channel = 0)
+    now = datetime.now()
+
+    #Formats data to json
+    fields = {"Date/Time": now, "Voltage": value/1023.0 * 3.3}
+    write_json(fields, log_file) 
+    print("Date: %s | Voltage: %.4f"% ( now ,(value/1023.0 * 3.3)))
+
+    # Delaysample
+    time.sleep(1)
```
Comment (imcewan7529) [inline adc_manager/src/adc.py:31]: We're targeting a sample rate of 10Hz, this should be sleep(.1)

File: adc_manager/src/test
Comment (imcewan7529) [inline adc_manager/src/test:1]: Can remove the blank 'test' files for this, they are just there as placeholders.

File: adc_manager/systemd/test
Comment (imcewan7529) [inline adc_manager/systemd/test:1]: Should add a service file here, look at https://vizartpandey.com/how-to-read-and-write-systemd/#:~:text=A%20.,process%20that%20systemd%20can%20manage for an example

File: adc_manager/src/MCP3008.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class MCP3008:
+	def __init__(self, bus = 0, device = 0):
+		self.bus, self.device = bus, device
+		self.spi = SpiDev()
+		self.open()
+		self.spi.max_speed_hz = 1350000 #1.35Mhz
+
+	def open(self):
+		self.spi.open(self.bus, self.device)
+		self.spi.max_speed_hz = 1350000 #1.35Mhz
```
Comment (imcewan7529) [inline adc_manager/src/MCP3008.py:12]: This is already set during __inti__, can it be removed here?

**Review by jdo1156 — COMMENTED** [review]
-added requirement.txt
-increased sampling rate
-uploaded log files

File: adc_manager/requirements/requirements.txt
Comment (imcewan7529) [inline adc_manager/requirements/requirements.txt:1]: I only see usage of one non-standard module in your code, SpiDev, don't think all these includes are needed

File: adc_manager/systemd/test
Comment (jdo1156) [reply adc_manager/systemd/test:1]: added service file

File: adc_manager/requirements/requirements.txt
Comment (jdo1156) [reply adc_manager/requirements/requirements.txt:1]: removed unnecessary requirement

File: adc_manager/src/MCP3008.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from spidev import SpiDev 
+
+class MCP3008:
+	def __init__(self, bus = 0, device = 0):
+		self.bus, self.device = bus, device
+		self.spi = SpiDev()
+		self.open()
```
Comment (imcewan7529) [inline adc_manager/src/MCP3008.py:7]: What does this line do?

File: adc_manager/src/adc_manager.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    #Formats data to json
+    fields = {"Date/Time": now, "Voltage": value/1023.0 * 3.3}
+    write_json(fields, log_file) 
+    print("Date: %s | Voltage: %.4f"% ( now ,(value/1023.0 * 3.3)))
+
+    # Delaysample
+    time.sleep(.1)
+
+log_file.close()
```
Comment (imcewan7529) [inline adc_manager/src/adc_manager.py:33]: Since this is preceded by a while true, it wont ever be executed

File: adc_manager/src/adc_manager.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from MCP3008 import MCP3008
+from datetime import datetime
+import json  
+import time
+
+
+adc = MCP3008()
+filename = '../test_logs/log_test.json'              # Name of log fie
```
Comment (imcewan7529) [inline adc_manager/src/adc_manager.py:8]: only /src/* will get copied into /usr/bin/, if you need a template .json, should put it in the /src/ directory

File: adc_manager/src/log_test.json
Comment (imcewan7529) [inline adc_manager/src/log_test.json:1]: Does this need to be in /src/?

File: adc_manager/src/test
Comment (imcewan7529) [inline adc_manager/src/test:1]: remove

File: adc_manager/src/adc_manager.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    while(True):
+        value = adc.read(channel = 0)/1023.0 * 5
+    
+        now = datetime.now(timezone.utc)
+        filtered_value = p1.Lowpass_Filter(value)
+        #Formats data to json
+        fields = {"Date/Time": now, "Voltage": filtered_value}
+        #write_json(fields, log_file) 
+        channel.basic_publish(exchange='',
+                      routing_key='queue_name',
```
Comment (imcewan7529) [inline adc_manager/src/adc_manager.py:51]: remove quotes around queue_name

**Review by imcewan7529 — APPROVED** [review]
LGTM, needs HW testing

**Pull request merged**